numpy/core/numerictypes.py

```diff
@@ -599,6 +600,16 @@ def find_common_type(array_types, scalar_types):
     """
     Determine common type following standard coercion rules.
 
+    .. deprecated:: NumPy 1.24
```

Comment:
Needs update to 1.25, also below. Might as well fix the long line while you are at it.